embedded-graphics/src/primitives/rectangle.rs

```diff
@@ -139,8 +138,7 @@ where
             top_left: self.top_left,
             bottom_right: self.bottom_right,
             style: self.style,
-            x: self.top_left[0],
-            y: self.top_left[1],
+            p: self.top_left,
```

Comment:
Could this be given a more descriptive name, or at least a doc comment explaining what it's for? It's got the same name in the other primitives also.

Reply:
Suggestions for a better name?

Reply:
It's the current point within the iterator, right? I don't mind keeping it at `p` for conciseness in the `next()` function, as long as there's a doc comment describing what it is. If you want to rename it, one of these perhaps?

* `current_point`
* `pos`
* `current_pos`
* `current_position`